src/features/defi/components/Deposit/Deposit.tsx

```diff
@@ -144,20 +146,33 @@ export const Deposit = ({
   )
 
   const handlePercentClick = useCallback(
-    (percent: number) => {
-      if (onPercentClick) onPercentClick(percent)
-      // The human crypto amount as a result of amount * percentage / 100, possibly with too many digits
-      const percentageCryptoAmount = bnOrZero(cryptoAmountAvailable).times(percent)
-      const percentageFiatAmount = percentageCryptoAmount.times(marketData.price)
+    async (percent: number) => {
+      const { percentageCryptoAmount, percentageFiatAmount } = await (async () => {
+        if (onPercentClick) {
+          const { amountCryptoPrecision: _amountCryptoPrecision, fiatAmount: _fiatAmount } =
+            await onPercentClick(percent)
+          return {
+            percentageCryptoAmount: bn(_amountCryptoPrecision),
+            percentageFiatAmount: _fiatAmount,
+          }
+        }
+
+        const _percentageCryptoAmount = bnOrZero(cryptoAmountAvailable).times(percent)
```

Comment:
we should probably use decimal places on all percent math at evaluation time so we don't accidentally set something to an overflow amount (ex. `CryptoAmount` field could have 19 decimal places instead of 18 currently, just view but still probs a good habit and then we don't have to have the extra `percentageCryptoAmountHuman` var)
```suggestion
        const _percentageCryptoAmount = bnOrZero(cryptoAmountAvailable).times(percent).dp(asset.precision)
```

Reply:
920f56bfa78d84bf96978cced5f571fc985f785f